### PR TITLE
[docs][dagster-dbt] add docs for row count metadata

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adds brief dbt reference section for `fetch_row_counts()` & adds apidocs for `DbtEventIterator`.

## Test Plan

Tested locally, vercel preview

### CODE REVIEW

**Comment by benpankow** [comment]
* **#21910** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21910?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/content/integrations/dbt/reference.mdx [hunk 1]
```diff
@@ -609,6 +609,41 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
 
 ---
 
+## Generate row count data for dbt models
```
Comment (rexledesma) [inline docs/content/integrations/dbt/reference.mdx:612]: Maybe this should be grouped under a section "Customizing asset materialization metadata".

Then, under that section, we should add "Fetch column-level lineage"

File: docs/content/integrations/dbt/reference.mdx [hunk 2]
```diff
@@ -609,6 +609,41 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
...
+
+<Note>
+  <strong>
+    Emitting row count data for dbt models is currently an experimental
+    feature.{" "}
+  </strong>{" "}
```
Comment (rexledesma) [inline docs/content/integrations/dbt/reference.mdx:618]: nit: just need one space.
```suggestion
    feature.
  </strong>{" "}
```

File: docs/content/integrations/dbt/reference.mdx (hunk 1 above)
Comment (rexledesma) [inline docs/content/integrations/dbt/reference.mdx:612]: for consistency with the API

```suggestion
## Fetch row count data for dbt models
```

File: docs/content/integrations/dbt/reference.mdx [hunk 3]
```diff
@@ -609,6 +609,41 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
...
+)
+def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+    yield from dbt.cli(["build"], context=context).stream().fetch_row_counts()
+```
+
+Once your dbt models have been materialized, you can view the row count data in the metadata table.
+
+---
+
 ## Emit column-level lineage as materialization metadata
```
Comment (rexledesma) [inline docs/content/integrations/dbt/reference.mdx:647]: Probably should also be "Fetch"

**Review by rexledesma — APPROVED** [review]
see nits on formating

File: docs/content/integrations/dbt/reference.mdx [hunk 4]
```diff
@@ -609,6 +609,41 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
...
+  <strong>
+    Emitting row count data for dbt models is currently an experimental
+    feature.{" "}
+  </strong>{" "}
+  To use this feature, you'll need to be on at least{" "}
+  <code>dagster>=0.17.6</code> and
+  <code>dagster-dbt>=0.23.6</code>.
+</Note>
+
+Dagster has the ability to automatically fetch row counts for dbt-generated tables and emit them as [materialization metadata](/concepts/metadata-tags/asset-metadata) to be displayed in the Dagster UI.
```
Comment (erinkcochran87) [inline docs/content/integrations/dbt/reference.mdx:624]:
```suggestion
Dagster can automatically fetch row counts for dbt-generated tables and emit them as [materialization metadata](/concepts/metadata-tags/asset-metadata) to be displayed in the Dagster UI.
```

**Review by erinkcochran87 — COMMENTED** [review]
A few small things, but looking good!

**Pull request merged**